LoopKit/Security/Security.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  Security.swift
+//  LoopKit
+//
+//  Created by Nathaniel Hamming on 2023-09-05.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public protocol SecurityPluginProvider {
+    var security: Security { get }
+}
+
+public protocol SecurityProvider {
+    /// The security with the specified identifier.
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the security
+    /// - Returns: Either a security with matching identifier or nil.
+    func security(withIdentifier identifier: String) -> Security?
+}
+
+public protocol Security: Pluggable {}
```

Comment:
I'm not seeing the need for this protocol. It doesn't specify any behavior.  I would assume that any behavior for a "Security" plugin would be specific to the type of security it's providing, and that its collaborators would have to access the specific implementation interface directly.

I.e. any PumpManager instances needing to access TidepoolSecurity will import `TidepoolSecurity`, get a `func initializationComplete(for plugins: [Plugable])` call, and look for an instance of `TidepoolSecurity` in there, and typecast to a local references for collaboration.

Reply:
I think the parameters to verify device and verify app will mostly likely differ based on the device and the service doing the verification, if one is involved, which is likely to be the case.

Reply:
The `Service` plugin does define some behavior, mainly that it has a delegate for persistence, lifecycle, issuing alerts, and supporting remote actions.

Reply:
As I'm implementing more, security is going to need the same persistence behaviour. Will be adding soon.